internal/plugin.go

```diff
@@ -424,10 +512,38 @@ func (p *Plugin) getPlrFromPath(plrPath string) (string, map[string]interface{},
 	var name string
 	var rule map[string]interface{}
 	for _, manifest := range *manifests {
-		if kind, _, _ := unstructured.NestedString(manifest, "kind"); kind != placementRuleKind {
+		kind, _, _ := unstructured.NestedString(manifest, "kind")
+		if kind != placementRuleKind && kind != placementKind {
 			continue
 		}
 
+		if kind == placementRuleKind && (!isPlR || p.plrCount.plr == 0) {
+			if !isPlR {
+				return "", nil, fmt.Errorf(
+					"the placement %s specified a placementRule kind but expected a placement kind", plrPath,
+				)
+			}
+			if p.plrCount.plr == 0 {
```

Comment:
The first check is whether a PlacementRule or Placement is expected. This second check is to ensure only one type of placement is used across the generated policies.

Reply:
I suppose I could make this clearer by instead of having the counter `plrCount` in the struct I could replace it with a `usingPlR` boolean and have the counts exist only in `assertValidConfig`.

Reply:
I like this approach!